autogen/versions.tf.tmpl

```diff
@@ -15,9 +15,15 @@
  */
 
 terraform {
-  required_version = "~> 0.12.6"
+  required_version = ">= 0.12"
   required_providers {
-    google      = ">= 3.32, <4.0.0"
-    google-beta = ">= 3.32, <4.0.0"
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 3.32, <4.0.0"
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
```

Comment:
I don't think we should make this jump until 0.13 is GA.
This syntax [will break](https://github.com/hashicorp/terraform/blob/0ab7c1b90e2e232ecf9519f891d849c7944abf3a/website/docs/configuration/provider-requirements.html.md#v012-compatible-provider-requirements) for tf < `v0.12.26`. Just relaxing the version constraint should be enough to support 0.13?

Reply:
It is going GA tomorrow https://github.com/hashicorp/terraform/issues/25016

Shall I leave it in there?

Reply:
Terraform 0.13 will not work without the new provider syntax. I think it is reasonable to expect users to be on the latest patch version.

Reply:
> Terraform 0.13 will not work without the new provider syntax. 

I have not tested this but other modules I am testing `v0.13.0-rc1` on seem to work fine without source as 0.13 defaults to `hashicorp` [namespace by default](https://github.com/hashicorp/terraform/blob/0ab7c1b90e2e232ecf9519f891d849c7944abf3a/website/docs/configuration/provider-requirements.html.md#v012-compatible-provider-requirements).

> I think it is reasonable to expect users to be on the latest patch version for 0.12
required_version = ">= 0.12"

I will defer to @morgante on this. If we do this, we should also set the constraint as such
```diff
+ required_version = ">= 0.12.26"
- required_version = ">= 0.12"
```

Reply:
I agree with @bharathkkb, we shouldn't force the new provider syntax until/unless need to. So let's stick to the minimal changes required for compatibility with 0.13—I expect it will be a while before everyone upgrades.

Reply:
I set it back to 0.12 version syntax